src/wasm/wasm-emscripten.cpp

```diff
@@ -697,10 +776,28 @@ void AsmConstWalker::queueImport(Name importName, std::string baseSig) {
   queuedImports.push_back(std::unique_ptr<Function>(import));
 }
 
+Literal AsmConstWalker::tableIndexForName(Name name) {
+  if (tableIndices.count(name)) {
+    return tableIndices[name];
+  }
+  assert(wasm.table.segments.size() == 1);
+  queuedTableEntries.push_back(name);
+  return tableIndices[name] = Literal(tableOffsets[0]++);
+}
+
 void AsmConstWalker::addImports() {
   for (auto& import : queuedImports) {
     wasm.addFunction(import.release());
   }
+
+  if (!queuedTableEntries.empty()) {
+    std::vector<Name>& tableSegmentData = wasm.table.segments[0].data;
+    assert(wasm.table.segments.size() == 1);
```

Comment:
Should move this before the `wasm.table.segments[0]`, because that technically depends on size being >= 1